source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -537,6 +634,86 @@ bool JsonTranscoderFilter::readToBuffer(Protobuf::io::ZeroCopyInputStream& strea
   return false;
 }
 
+void JsonTranscoderFilter::createHttpBodyEnvelope(Buffer::Instance& data, uint64_t content_length) {
```

Comment:
I feel this code is grpc specific,  can be moved to grpc util library.  it should be unit-tested thoroughly.

Reply:
Few concerns:
1. It's more protobuf than grpc specific (grpc piece is small and is not a concern)
2. I'm not sure it's useful anywhere outside of this filter

I agree though that it's not trivial and I'm adding better test coverage

Reply:
> Few concerns:
> 
> 1. It's more protobuf than grpc specific (grpc piece is small and is not a concern)
> 2. I'm not sure it's useful anywhere outside of this filter
> 
> I agree though that it's not trivial and I'm adding better test coverage

This is both grpc and protobuf specific because of the grpc framing part. Ideally it should be in the transcoding library, and a fuzz test should be added too.

Reply:
gRPC-specific code was moved to shared library so it's re-used with http1 reverse bridge.
HttpBody protobuf-specific code was covered by tests for the edge cases.

I'm not sure what would be a benefit of a fuzz testing in this case (user input is limited and well-defined).